akka-typed-testkit/src/main/scala/akka/typed/testkit/Effects.scala

```diff
@@ -79,16 +79,21 @@ class EffectfulActorContext[T](_name: String, _initialBehavior: Behavior[T], _ma
 
   override def spawnAnonymous[U](behavior: Behavior[U], props: Props = Props.empty): ActorRef[U] = {
     val ref = super.spawnAnonymous(behavior, props)
-    effectQueue.offer(Spawned(ref.path.name))
+    effectQueue.offer(Spawned(ref.path.name, Option(props)))
     ref
   }
-  override def spawnAdapter[U](f: U ⇒ T, name: String = ""): ActorRef[U] = {
```

Comment:
Due to conflicts with other methods in the hierarchy, when no `name` was provided this method was never called, but the `spawnAdapter(f)` defined upper in the hierarchy. To solve this, I override here both variants with and without name.